[DROOLS-799] fix update-version-all.sh script

psiroky opened the pull request:
* script now automatically updates all the required
   versions in order to successfully build all the repos

 * removed the call to ant script in buid-distribution
   at the end; using simple sed string replacements instead

 * the current (old) version param is not used, so no need
   to pass it in

@mbiarnes please take a look and let me know if you have any questions. I tested the script locally, but would be great if you could test it as well to be sure everything works as expected.

psiroky commented:
Some of the POMs (and other files) are not being updated when running the script. Mostly becuase they are not part of the Maven build (they are not being released). We need to figure out what to do with them. Either remote the modules altogether or update the versions manually (sed).

List of the files:
`drools/kie-maven-plugin-example/pom.xml`
`droolsjbpm-integration/kie-remote/kie-remote-ws/kie-remote-ws-gen-wsdl/pom.xml`
`drools-wb/drools-wb-jcr2vfs-migration/drools-wb-jcr2vfs-migration-core/pom.xml`
`jbpm/jbpm-human-task/jbpm-human-task-performance/pom.xml`
`jbpm/jbpm-installer/build.properties:snapshot.version=6.3.0-SNAPSHOT`
`jbpm/jbpm-installer/build.properties:release.version=6.3.0-SNAPSHOT`
`jbpm-designer/jbpm-designer-utilities/pom.xml`
`kie-wb-distributions/kie-docs/droolsjbpm-introduction-docs/pom.xml`
`kie-wb-distributions/kie-docs/shared/Spring/KieSpring.xml`

mbiarnes commented:
Hi Petr,

I will test it today.

Michael

----- Ursprüngliche Mail -----
Von: "Petr Široký" <notifications@github.com>
An: "droolsjbpm/droolsjbpm-build-bootstrap" <droolsjbpm-build-bootstrap@noreply.github.com>
CC: "mbiarnes" <mbiarnes@redhat.com>
Gesendet: Freitag, 29. Mai 2015 11:58:16
Betreff: [droolsjbpm-build-bootstrap] [DROOLS-799] fix update-version-all.sh script (#71)

 * script now automatically updates all the required
   versions in order to successfully build all the repos

 * removed the call to ant script in buid-distribution
   at the end; using simple sed string replacements instead

 * the current (old) version param is not used, so no need
   to pass it in

@mbiarnes please take a look and let me know if you have any questions. I tested the script locally, but would be great if you could test it as well to be sure everything works as expected.
You can view, comment on, or merge this pull request online at:

  https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/71

-- Commit Summary --

  * [DROOLS-799] fix update-version-all.sh script

-- File Changes --

    M script/release/update-version-all.sh (68)

-- Patch Links --

https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/71.patch
https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/71.diff

---
Reply to this email directly or view it on GitHub:
https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/71

psiroky commented:
I removed some of the old modules and fixed some other references. The following are remaining files that need to be fixed:

`drools/kie-maven-plugin-example/pom.xml`
`jbpm/jbpm-human-task/jbpm-human-task-performance/pom.xml`
`jbpm-designer/jbpm-designer-utilities/pom.xml`

jlindop commented:
Hi Petr, 

I know about jbpm-designer/jbpm-designer-utilities/pom.xml and as you know, it's not updated because it's not included in the jbpm-designer pom. Maybe the best way to get update-version-all.sh to update the versions for these files is to make the script update the version directly. What do you think? 

Jeremy 

----- Original Message -----

From: "Petr Široký" <notifications@github.com> 
To: "droolsjbpm/droolsjbpm-build-bootstrap" <droolsjbpm-build-bootstrap@noreply.github.com> 
Sent: Tuesday, 2 June, 2015 8:42:16 AM 
Subject: Re: [droolsjbpm-build-bootstrap] [DROOLS-799] fix update-version-all.sh script (#71) 



I removed some of the old modules and fixed some other references. The following are remaining files that need to be fixed: 

drools/kie-maven-plugin-example/pom.xml 
jbpm/jbpm-human-task/jbpm-human-task-performance/pom.xml 
jbpm-designer/jbpm-designer-utilities/pom.xml 

— 
Reply to this email directly or view it on GitHub .

mbiarnes commented:
Hi Petr. hi Jeremy,

This is what I did exactly for the releases. as modules weren't build it did not hurt also to change the -SNAPSHOTs to a version. So I get rid of these SNAPSHOTs

psiroky commented:
I agree that for these special cases the script should just update the version directly. I'll look into that.

mbiarnes commented:
Checked. Works perfect. There are still some SNAPSHOTs not updated automatically as there are modules that are not build - but I think this is OK. We can fix this in future.
+1 to merge this PR

psiroky commented:
Merged into master https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/commit/586dc1b16a3fbc9e0259ca22d03e7005140ed1ac . The leftovers will be fixed in near future.
